lib/fastlane/plugin/revenuecat_internal/helper/versioning_helper.rb

```diff
@@ -0,0 +1,145 @@
+          else
+            UI.user_error!("Cannot determine next version. Multiple commits found for #{sha}")
+          end
+        end
+        UI.important("Type of bump after version #{old_version} is #{type_of_bump}")
+        increase_version(old_version, type_of_bump, false)
+      end
+
+      def self.auto_generate_changelog(repo_name, github_token, rate_limit_sleep)
+        old_version = Actions.sh("git describe --tags --abbrev=0").strip
+        UI.important("Auto-generating changelog since #{old_version}")
+
+        commits = Helper::GitHubHelper.get_commits_since_old_version(github_token, old_version, repo_name)
+
+        changelog_sections = { breaking_changes: [], fixes: [], new_features: [], other: [] }
+
+        commits.map do |commit|
+          name = commit["commit"]["author"]["name"]
+
+          sha = commit["sha"]
+          items = Helper::GitHubHelper.get_pr_resp_items_for_sha(sha, github_token, rate_limit_sleep, repo_name)
+
+          if items.size == 1
+            item = items.first
+
+            message = "#{item['title']} (##{item['number']})"
+            username = item["user"]["login"]
+            types_of_change = get_type_of_change_from_pr_info(item)
+
+            section = get_section_depending_on_types_of_change(types_of_change)
+
+            line = "* #{message} via #{name} (@#{username})"
+            changelog_sections[section].push(line)
+          else
+            UI.user_error!("Cannot generate changelog. Multiple commits found for #{sha}")
+          end
+        end
+        build_changelog_sections(changelog_sections)
+      end
+
+      private_class_method def self.build_changelog_sections(changelog_sections)
+        changelog_sections.reject { |_, v| v.empty? }.map do |section_name, prs|
+          next unless prs.size > 0
+
+          case section_name
+          when :breaking_changes
+            title = "### Breaking Changes"
+          when :fixes
+            title = "### Bugfixes"
+          when :new_features
+            title = "### New Features"
+          else
+            title = "### Other Changes"
+          end
+          "#{title}\n#{prs.join("\n")}"
+        end.join("\n")
+      end
+
+      private_class_method def self.get_section_depending_on_types_of_change(change_types)
+        if change_types.include?("breaking")
+          :breaking_changes
+        elsif change_types.include?("feat")
+          :new_features
+        elsif change_types.include?("fix")
+          :fixes
+        else
+          :other
+        end
+      end
+
+      private_class_method def self.get_type_of_bump_from_types_of_change(change_types)
+        if change_types.include?("breaking")
+          :major
+        elsif change_types.include?("feat")
+          :minor
+        else
+          :patch
+        end
+      end
+
+      def self.increase_version(current_version, type_of_bump, snapshot)
```

Comment:
Nitpick, I would move public methods above private ones.

Reply:
Also, I believe we need to add tests for this function, even though it's already tested through some of the usages. I would keep a test for those usages and test it more thoroughly in the tests for the helper.